indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -567,19 +581,26 @@ public Response getCompleteTasks(
         )
     );
 
-    final List<TaskStatusPlus> completeTasks = recentlyFinishedTasks
-        .stream()
-        .map(status -> new TaskStatusPlus(
-            status.getId(),
-            taskFunction.apply(status.getId()).getType(),
-            taskStorageQueryAdapter.getCreatedTime(status.getId()),
-            // Would be nice to include the real queue insertion time, but the TaskStorage API doesn't yet allow it.
-            DateTimes.EPOCH,
-            status.getStatusCode(),
-            status.getDuration(),
-            TaskLocation.unknown())
-        )
-        .collect(Collectors.toList());
+    final Function<TaskStatus, Pair<DateTime, String>> cTDsFunction = taskStatus -> {
+      return taskStorageQueryAdapter.getCreatedDateAndDataSource(taskStatus.getId());
+    };
+    
+    final List<TaskStatusPlus> completeTasks = Lists.newArrayList(Iterables.transform(
+        recentlyFinishedTasks,
+        status -> {
+          final Pair<DateTime, String> pair = cTDsFunction.apply(status);
```

Comment:
minor nit: you can directly call taskStorageQueryAdapter.get instead of creating cTDsFunction

Reply:
Thanks. Fixed the same